colab/plugins/utils/apps.py

```diff
@@ -1,6 +1,32 @@
+import importlib
+import inspect
 
 from django.apps import AppConfig
+from colab.plugins.utils.signals import AbstractSignal
 
 
 class ColabProxiedAppConfig(AppConfig):
     colab_proxied_app = True
+
+    def __init__(self, app_name, app_module):
+        super(ColabProxiedAppConfig, self).__init__(app_name, app_module)
+        self._import_signals(app_name)
+        self.signals.register_signal()
+
+
+    def _import_signals(self, app_name):
```

Comment:
How can we do the same but with less magic? 

Django had a very big refactor before it got popular. Their refactor was called "removing the magic": https://code.djangoproject.com/wiki/RemovingTheMagic). Just as a reference.